code/game/machinery/belts/launcherloaders.dm

```diff
@@ -0,0 +1,186 @@
+/obj/machinery/launcher_loader
+	icon = 'goon/icons/obj/stationobjs.dmi'
+	icon_state = "launcher_loader_0"
+	name = "Automatic mass-driver loader (AMDL)"
+	desc = "An automated, hydraulic mass-driver loader."
+	density = 0
+	opacity = 0
+	layer = 2.6
+	anchored = 1
+
+	var/obj/machinery/mass_driver/driver = null
+
+	var/id = "null"
+	var/operating = 0
+	var/driver_operating = 0
+	var/trash = 0
+	var/obj/machinery/door/poddoor/door = null
+
+/obj/machinery/launcher_loader/New()
+	..()
+	spawn(5)
+		var/list/drivers = new/list()
+		for(var/obj/machinery/mass_driver/D in range(1,src))
+			drivers += D
+		if(drivers.len)
+			if(drivers.len > 1)
+				for(var/obj/machinery/mass_driver/D2 in drivers)
+					if(D2.id == src.id)
+						driver = D2
+						break
+				if(!driver) driver = pick(drivers)
+			else
+				driver = pick(drivers)
+			src.dir = get_dir(src,driver)
+
+/obj/machinery/launcher_loader/proc/destroy()
+	if (driver)
+		driver = null
+	if (door)
+		door = null
+	return ..()
+
+/obj/machinery/launcher_loader/proc/activate()
+	if(operating || !isturf(src.loc)) return
+	operating = 1
+	flick("launcher_loader_1",src)
+	playsound(src, "goon/sound/effects/pump.ogg",50, 1)
+	sleep(3)
+	for(var/atom/movable/AM in src.loc)
+		if(AM.anchored || AM == src) continue
+		if(trash && AM.delivery_destination != "Disposals")
+			AM.delivery_destination = "Disposals"
+		step(AM,src.dir)
+	operating = 0
+	handle_driver()
+
+/obj/machinery/launcher_loader/proc/handle_driver()
+	if(driver && !driver_operating)
+		driver_operating = 1
+
+	spawn(0)
+		for(var/obj/machinery/door/poddoor/P)
```

Comment:
is this  a loop over contents? This doesn't even have a list associated, how the fuck does this code even work